### PR TITLE
added example how to add username and password to a request

### LINKED ISSUE #78: Add example of how to add credentials to request to README
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

Add a simple example of how to add credentials to a request, to make it easier for beginners to use the plugin.

It's my first Issue so im sorry if i did something wrong

### Motivation

it is unclear, where this plugin gets its the username and password from, if one is not familiar with how authorisation works

## How to add Credentials to request
The basic-auth plugin retrieves the credentials from the [Authorization](https://en.wikipedia.org/wiki/Basic_access_authentication) header.

```typescript
        const credentials: string = Buffer.from("username:password").toString('base64')
       request.header("Authorisazion", "Basic " + credentials)
```

### PR DESCRIPTION
**Pull request by gertminov**
resolves #78 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

File: README.md [hunk 1]
```diff
@@ -99,6 +99,30 @@ fastify.after(() => {
 })
 ```
 
+### How to add credentials to a request
+The basic-auth plugin retrieves the credentials from the [Authorization](https://en.wikipedia.org/wiki/Basic_access_authentication) header.
+
+A simple example to add `username` and `password` to the reply header:
```
Comment (jsumners) [inline README.md:105]: Why would we give an example showing sending the credentials in a response header? https://www.rfc-editor.org/rfc/rfc7617.html does not call for this, and it seems like a bad practice to recommend.

**Review by jsumners — COMMENTED** [review]
I am unclear about the response example, and I think the plugin is pretty clear that it implements basic authentication (i.e. https://www.rfc-editor.org/rfc/rfc7617.html). So it seems redundant to me to add an example, but I suppose it doesn't hurt anything either. I would just remove the "send credentials back the client" portion as it isn't spec compliant and is a glaring security violation.

File: README.md [hunk 2]
```diff
@@ -99,6 +99,30 @@ fastify.after(() => {
...
+fastify.get('/',  (request, reply) =>{
+    const credentials: string = Buffer.from("username:password").toString('base64')
+    reply.header("Authorisazion", "Basic " + credentials)
+    reply.send()
+})
+```
+
+An example to add `username` and `password` via [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch):
+```typescript
+fetch('URL_GOES_HERE', {
```
Comment (Eomm) [inline README.md:117]:
```suggestion
fetch('https://www.example.org/', { 
```

**Review by lilpoozie2005 — APPROVED** [review]

**Pull request closed without merging**